src/app/components/Root/Root.tsx

```diff
@@ -3,16 +3,47 @@ import './Root.css'
 import { SearchResults, SearchBar } from "../../Recipes";
 import { NavBar } from "../NavBar";
 import { WeatherWidget } from "../../Weather";
+import { ThemeName } from "../../types";
+import { themes } from "../../constants";
+
+export type RootState = {
+    themeName: ThemeName;
+}
 
 export class Root extends React.Component {
+    state: RootState = {
+        themeName: 'dark'
+    };
+
+    toggleTheme = () => {
+        this.setState(() => ({
+            themeName: this.state.themeName === 'light' ? 'dark' : 'light'
+        }))
+    };
+
     public render() {
+        const { Provider, Consumer } = React.createContext(this.state.themeName);
+        const { themeName } = this.state;
+
         return (
-            <div className="root">
-                <NavBar/>
-                <SearchBar/>
-                <WeatherWidget />
-                <SearchResults/>
-            </div>
+          <Provider value={themeName}>
+              <div className="root">
+                  <Consumer>
```

Comment:
the value in context is the implicit passing of props. in this case, `<NavBar />`'s props don't need to know about the theme. `<NavBar />` should render the `<Consumer />`. As of right now, this component shouldn't need to render a `<Consumer />`